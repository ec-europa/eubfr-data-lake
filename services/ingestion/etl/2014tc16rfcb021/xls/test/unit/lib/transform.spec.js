/**
 * @jest-environment node
 */

import mapper from '../../../src/lib/transform';
import testRecord from '../../stubs/record.json';
import testRecordDates from '../../stubs/recordDates.json';

describe('2014tc16rfcb021 XLS transformer', () => {
  let result = {};
  let resultDates = {};

  beforeAll(() => {
    result = mapper(testRecord);
    resultDates = mapper(testRecordDates);
  });

  test('Returns null when record is not provided', () => {
    expect(mapper()).toBe(null);
  });

  test('Produces correct JSON output structure', () => {
    expect(result).toMatchSnapshot();
  });

  test('Formats dates only if in the form of DD.MM.YYYY', () => {
    // Make a change in the copy, no need for another stub for testing a specific field.
    const copy = JSON.parse(JSON.stringify(testRecord));
    // Change one of the dates to not be of "DD.MM.YYYY", there shouldn't be any change in the output.
    copy['End date'] = '2018-11-28T21:00:00.000Z';
    const resultCopy = mapper(copy);
    expect(resultCopy.timeframe.from).toBe('2016-02-03T00:00:00.000Z');
    expect(resultCopy.timeframe.to).toBe('2018-11-28T21:00:00.000Z');
  });

  test('Produces correct JSON output in case of Dates in an ISO format', () => {
    console.log(resultDates);
    expect(resultDates).toMatchSnapshot();
  });
});
