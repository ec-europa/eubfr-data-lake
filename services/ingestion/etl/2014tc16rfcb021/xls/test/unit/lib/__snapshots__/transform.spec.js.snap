// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`2014tc16rfcb021 XLS transformer Produces correct JSON output in case of Dates in an ISO format 1`] = `
Object {
  "action": "",
  "budget": Object {
    "eu_contrib": Object {
      "currency": "EUR",
      "raw": 6660192.21,
      "value": 6660192.21,
    },
    "funding_area": Array [],
    "mmf_heading": "",
    "other_contrib": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "private_fund": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "public_fund": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "total_cost": Object {
      "currency": "EUR",
      "raw": 7835520.26,
      "value": 7835520.26,
    },
  },
  "call_year": "",
  "complete": false,
  "description": "Ranking: 13 
Project code: N/A 
Objectives: Objective: modernizing the road infrastructure that is part of TEN-T network or that represents direct road connections to TEN-T by modernizing 6 662 m of infrastructure in Giurgiu Municipality, and 4 300 m in Ruse. 
Duration: 30 months, 1 day 
Category of intervention: 34 
Percent (ERDF): 0.85 
National co-financing(euro): 1018539.3 
Percent (National co-financing): 0.13 
Own Contribution (euro): 156788.75 
Percent (Own Contributions): 0.02 
",
  "ec_priorities": Array [],
  "media": Array [],
  "programme_name": "",
  "project_id": "2f303693f960d7171230ae2664a8302b",
  "project_locations": Array [
    Object {
      "address": "",
      "centroid": null,
      "country_code": "RO",
      "location": null,
      "nuts": Array [],
      "postal_code": "",
      "region": "Giurgiu",
      "town": "",
    },
  ],
  "project_website": "",
  "related_links": Array [],
  "reporting_organisation": "Member states",
  "results": Object {
    "available": "",
    "result": "",
  },
  "status": "In implementation",
  "sub_programme_name": "",
  "success_story": "",
  "themes": Array [],
  "third_parties": Array [
    Object {
      "address": "",
      "country": "Romania",
      "email": "",
      "name": "Territorial Administrative Unit - Giurgiu Town",
      "phone": "",
      "region": "Giurgiu",
      "role": "coordinator",
      "type": "",
      "website": "",
    },
  ],
  "timeframe": Object {
    "from": "2018-09-07T00:00:00.000Z",
    "from_precision": "day",
    "to": "2021-03-07T00:00:00.000Z",
    "to_precision": "day",
  },
  "title": "Well-developed transportation system in the Euroregion Ruse-Giurgiu for better connectivity with TEN-T network",
  "type": Array [],
}
`;

exports[`2014tc16rfcb021 XLS transformer Produces correct JSON output structure 1`] = `
Object {
  "action": "",
  "budget": Object {
    "eu_contrib": Object {
      "currency": "EUR",
      "raw": 1214450.87,
      "value": 1214450.87,
    },
    "funding_area": Array [],
    "mmf_heading": "",
    "other_contrib": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "private_fund": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "public_fund": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "total_cost": Object {
      "currency": "EUR",
      "raw": 1428765.73,
      "value": 1428765.73,
    },
  },
  "call_year": "",
  "complete": false,
  "description": "Ranking: 1 
Project code: 15.1.1.010 
Objectives: Objective: to significantly improve the planning, development and coordination of CBC transport systems for better connections with TEN-T network in CBC area                            
 
Duration: 24 
Category of intervention: 44 
Percent (ERDF): 0.85 
National co-financing(euro): 185739.5449 
Percent (National co-financing): 0.13 
Own Contribution (euro): 28575.3146 
Percent (Own Contributions): 0.02 
",
  "ec_priorities": Array [],
  "media": Array [],
  "programme_name": "",
  "project_id": "b1c6f005bfb0dff460a6f0331c7a1af2",
  "project_locations": Array [
    Object {
      "address": "",
      "centroid": null,
      "country_code": "BG",
      "location": null,
      "nuts": Array [],
      "postal_code": "",
      "region": "Pleven",
      "town": "",
    },
  ],
  "project_website": "",
  "related_links": Array [],
  "reporting_organisation": "Member states",
  "results": Object {
    "available": "",
    "result": "",
  },
  "status": "Finalized",
  "sub_programme_name": "",
  "success_story": "",
  "themes": Array [],
  "third_parties": Array [
    Object {
      "address": "",
      "country": "Bulgaria ",
      "email": "",
      "name": "Association of Danube River Municipalities \\"Danube\\" (ADRM)",
      "phone": "",
      "region": "Pleven",
      "role": "coordinator",
      "type": "",
      "website": "",
    },
  ],
  "timeframe": Object {
    "from": "2016-02-03T00:00:00.000Z",
    "from_precision": "day",
    "to": "2018-02-02T00:00:00.000Z",
    "to_precision": "day",
  },
  "title": "Investigation of opportunities for reducing the TEN-T network use within the cross-border region of Romania-Bulgaria through optimazition of the freight and passanger transport and the development of a joint mechanism foe support of the intermodal connection",
  "type": Array [],
}
`;
